src/__tests__/helpers.ts

```diff
@@ -3,12 +3,21 @@ import {
     Serializer,
     serializers as predefinedSerializers,
 } from '../';
+import { resetUuidCounter } from './setup';
 
-type SerializerType = 'json' | 'yaml' | 'dot' | 'structure';
+type SerializerType =
```

Comment:
Could this be an option for a generic `Serializer`? Like:

```
await new Serializer({ type: 'yaml_depends_on' }).serialize(pipeline)
```

Reply:
I removed the specialized serializers completely now after @uri-canva's request. It's an option on each of the serializers and I introduced a new type for it.